neuralmonkey/encoders/conv_s2s_encoder.py

```diff
@@ -0,0 +1,189 @@
+"""From the paper Convolutional Sequence to Sequence Learning
+
+http://arxiv.org/abs/1705.03122
+"""
+
+import tensorflow as tf
+import numpy as np
+from typeguard import check_argument_types
+
+from neuralmonkey.encoders.attentive import Attentive
+from neuralmonkey.model.model_part import ModelPart, FeedDict
+from neuralmonkey.logging import log
+from neuralmonkey.dataset import Dataset
+from neuralmonkey.decorators import tensor
+from neuralmonkey.nn.projection import glu, linear
+from neuralmonkey.nn.utils import dropout
+from neuralmonkey.vocabulary import Vocabulary
+from neuralmonkey.model.sequence import (EmbeddedSequence)
+
+
+class ConvolutionalSentenceEncoder(ModelPart, Attentive):
+
+    # pylint: disable=too-many-arguments
+    def __init__(self,
+                 name: str,
+                 input_sequence: EmbeddedSequence,
+                 conv_features: int,
+                 encoder_layers: int,
+                 kernel_width: int = 5,
+                 dropout_keep_prob: float = 1.0,
+                 attention_type: type = None,
+                 attention_state_size: int = None,
+                 attention_fertility: int = 3,
+                 save_checkpoint: str = None,
+                 load_checkpoint: str = None) -> None:
+
+        ModelPart.__init__(self, name, save_checkpoint, load_checkpoint)
+        Attentive.__init__(self, attention_type,
+                           attention_state_size=attention_state_size,
+                           attention_fertility=attention_fertility)
+
+        assert check_argument_types()
```

Comment:
tady nepatří ten `assert`. A celý to patří nad ty volání těch konstruktorů. Jestli je to jinde jinak (jako že nejspíš je), tak je to špatně tam. Konstruktor modelpartu si totiž ty typy už nekontroluje.